[CI] ExplainLifecycleResponseTests testConcurrentToXContent failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/bnz5qupzg3sy6/tests/:x-pack:plugin:core:test/org.elasticsearch.xpack.core.ilm.ExplainLifecycleResponseTests/testConcurrentToXContent


**Reproduction line:**
```
./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.ilm.ExplainLifecycleResponseTests.testConcurrentToXContent" -Dtests.seed=CDC9BA8EC8F802C9 -Dtests.locale=mt-MT -Dtests.timezone=Etc/GMT-3 -Druntime.java=11 -Dtests.fips.enabled=true
```

**Applicable branches:**
7.17

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.core.ilm.ExplainLifecycleResponseTests#testConcurrentToXContent`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testConcurrentToXContent'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.core.ilm.ExplainLifecycleResponseTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: Failed to round trip over human readable YAML
Canonical is:
{
  "indices" : {
    "ExrMVYkezr" : {
      "index" : "ExrMVYkezr",
      "managed" : true,
      "policy" : "gcfWNXpoDI",
      "lifecycle_date" : "1984-01-26T14:59:41.124Z",
      "lifecycle_date_millis" : 443977181124,
      "age" : "14607.29d",
      "phase" : "JOZmXqIJWA",
      "phase_time" : "110265758-11-11T13:17:56.195Z",
      "phase_time_millis" : 3479589092467076195,
      "action" : "UHmIbQJFlj",
      "action_time" : "9269006-03-22T13:46:30.287Z",
      "action_time_millis" : 292439417175990287,
      "step" : "fvNXNNrNXm",
      "step_time" : "43374862-06-29T06:36:02.846Z",
      "step_time_millis" : 1368716286494162846,
      "failed_step" : "QFqKMjbGgB",
      "is_auto_retryable_error" : false,
      "failed_step_retry_count" : 6,
      "repository_name" : "asAfqdlbBU",
      "snapshot_name" : "CSPZgTSDNW",
      "shrink_index_name" : "OsCSHbzVyF",
      "step_info" : {
        "SUgDjaQTBm" : "KHyaRkHRJD"
      },
      "phase_execution" : {
        "policy" : "FmDpY",
        "phase_definition" : {
          "min_age" : "721280350d",
          "actions" : {
            "TEST_ACTION" : { }
          }
        },
        "version" : 9127859814447364793,
        "modified_date" : "221011433-12-08T04:22:09.389Z",
        "modified_date_in_millis" : 6974385044934129389
      }
    },
    "IpqkNuAUIr" : {
      "index" : "IpqkNuAUIr",
      "managed" : true,
      "policy" : "aAEcNKfMcT",
      "lifecycle_date" : "2021-07-19T05:16:39.090Z",
      "lifecycle_date_millis" : 1626671799090,
      "age" : "918.7d",
      "failed_step" : "PpfECoZKLX",
      "phase_execution" : {
        "policy" : "zMRHu",
        "phase_definition" : {
          "min_age" : "0ms",
          "actions" : {
            "TEST_ACTION" : { }
          }
        },
        "version" : 1250135226093044054,
        "modified_date" : "52182977-11-09T15:24:26.568Z",
        "modified_date_in_millis" : 1646673560292266568
      }
    }
  }
}
Wanted : ---
indices:
  ExrMVYkezr:
    index: "ExrMVYkezr"
    managed: true
    policy: "gcfWNXpoDI"
    lifecycle_date: "1984-01-26T14:59:41.124Z"
    lifecycle_date_millis: 443977181124
    age: "14607.29d"
    phase: "JOZmXqIJWA"
    phase_time: "110265758-11-11T13:17:56.195Z"
    phase_time_millis: 3479589092467076195
    action: "UHmIbQJFlj"
    action_time: "9269006-03-22T13:46:30.287Z"
    action_time_millis: 292439417175990287
    step: "fvNXNNrNXm"
    step_time: "43374862-06-29T06:36:02.846Z"
    step_time_millis: 1368716286494162846
    failed_step: "QFqKMjbGgB"
    is_auto_retryable_error: false
    failed_step_retry_count: 6
    repository_name: "asAfqdlbBU"
    snapshot_name: "CSPZgTSDNW"
    shrink_index_name: "OsCSHbzVyF"
    step_info:
      SUgDjaQTBm: "KHyaRkHRJD"
    phase_execution:
      policy: "FmDpY"
      phase_definition:
        min_age: "721280350d"
        actions:
          TEST_ACTION: {}
      version: 9127859814447364793
      modified_date: "221011433-12-08T04:22:09.389Z"
      modified_date_in_millis: 6974385044934129389
  IpqkNuAUIr:
    index: "IpqkNuAUIr"
    managed: true
    policy: "aAEcNKfMcT"
    lifecycle_date: "2021-07-19T05:16:39.090Z"
    lifecycle_date_millis: 1626671799090
    age: "918.69d"
    failed_step: "PpfECoZKLX"
    phase_execution:
      policy: "zMRHu"
      phase_definition:
        min_age: "0ms"
        actions:
          TEST_ACTION: {}
      version: 1250135226093044054
      modified_date: "52182977-11-09T15:24:26.568Z"
      modified_date_in_millis: 1646673560292266568

But got: ---
indices:
  ExrMVYkezr:
    index: "ExrMVYkezr"
    managed: true
    policy: "gcfWNXpoDI"
    lifecycle_date: "1984-01-26T14:59:41.124Z"
    lifecycle_date_millis: 443977181124
    age: "14607.29d"
    phase: "JOZmXqIJWA"
    phase_time: "110265758-11-11T13:17:56.195Z"
    phase_time_millis: 3479589092467076195
    action: "UHmIbQJFlj"
    action_time: "9269006-03-22T13:46:30.287Z"
    action_time_millis: 292439417175990287
    step: "fvNXNNrNXm"
    step_time: "43374862-06-29T06:36:02.846Z"
    step_time_millis: 1368716286494162846
    failed_step: "QFqKMjbGgB"
    is_auto_retryable_error: false
    failed_step_retry_count: 6
    repository_name: "asAfqdlbBU"
    snapshot_name: "CSPZgTSDNW"
    shrink_index_name: "OsCSHbzVyF"
    step_info:
      SUgDjaQTBm: "KHyaRkHRJD"
    phase_execution:
      policy: "FmDpY"
      phase_definition:
        min_age: "721280350d"
        actions:
          TEST_ACTION: {}
      version: 9127859814447364793
      modified_date: "221011433-12-08T04:22:09.389Z"
      modified_date_in_millis: 6974385044934129389
  IpqkNuAUIr:
    index: "IpqkNuAUIr"
    managed: true
    policy: "aAEcNKfMcT"
    lifecycle_date: "2021-07-19T05:16:39.090Z"
    lifecycle_date_millis: 1626671799090
    age: "918.7d"
    failed_step: "PpfECoZKLX"
    phase_execution:
      policy: "zMRHu"
      phase_definition:
        min_age: "0ms"
        actions:
          TEST_ACTION: {}
      version: 1250135226093044054
      modified_date: "52182977-11-09T15:24:26.568Z"
      modified_date_in_millis: 1646673560292266568

  at org.junit.Assert.fail(Assert.java:89)
  at org.elasticsearch.test.AbstractSerializingTestCase.lambda$testConcurrentToXContent$3(AbstractSerializingTestCase.java:98)
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
  at java.util.concurrent.FutureTask.run(FutureTask.java:264)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
  at java.lang.Thread.run(Thread.java:834)

```

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/ExplainLifecycleResponseTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/ExplainLifecycleResponseTests.java
@@ -26,8 +26,11 @@ public class ExplainLifecycleResponseTests extends AbstractXContentSerializingTe
     @Override
     protected ExplainLifecycleResponse createTestInstance() {
         Map<String, IndexLifecycleExplainResponse> indexResponses = new HashMap<>();
+        long now = System.currentTimeMillis();
         for (int i = 0; i < randomIntBetween(0, 2); i++) {
             IndexLifecycleExplainResponse indexResponse = IndexLifecycleExplainResponseTests.randomIndexExplainResponse();
+            // Since the age is calculated from now, we make now constant so that we don't get changes in age during the run of the test:
+            indexResponse.nowSupplier = () -> now;
             indexResponses.put(indexResponse.getIndex(), indexResponse);
         }
         return new ExplainLifecycleResponse(indexResponses);

